projects/example-app/src/app/auth/actions/auth-api.actions.ts

```diff
@@ -1,20 +1,17 @@
-import { props, createAction } from '@ngrx/store';
+import { createAction, props, union } from '@ngrx/store';
 import { User } from '@example-app/auth/models/user';
 
-export const loginSuccess = createAction(
-  '[Auth/API] Login Success',
-  props<{ user: User }>()
-);
+export const AuthApiActions = {
```

Comment:
I was think to add the docs myself as well 😛
Anyhow, that's not quite how I see new action creators to be used for a few reasons:
1) I think "module import" works well without the need to wrap with extra object.
2) Google's TS style guide says "When providing a structural-based implementation, explicitly include the type at the declaration of the symbol". So we won't be able to follow the pattern.
3) I'm thinking to change the union fn to take an array instead. That's what @cartant did with `ts-action` and it feels like passing a collection makes more sense from semantics perspective.
4) Didn't verify this personally, but I think unused action creators would be easier to "tree-shake" when they are by themselves (instead of part of the object - that too might be already working, but could depend on the code optimizer).

Reply:
Good points. That's why I left this as a separate commit.

1. I'm thinking about it more in terms of ergonomics. I think having 1 export to use across reducers and effects rather than TheseActions and TheseActionsUnion is nice. Idea about declaration merging from here https://medium.com/@martin_hotell/improved-redux-type-safety-with-typescript-2-8-2c11a8062575
2. Unless you're talking about schematics, I don't see how this prevents Google from following that pattern, and is common that you build apps in a different style either way 😉. 
3. Ok
4. Ok

Reply:
1. Even from ergonomics.
    -  Files like projects/example-app/src/app/auth/actions/index.ts become unnecessary (we actually never used them anyway)
    - name the union as `All` and leave action creators as is.
    - Import in the reducer file as
    ```ts
    import * as authApiActions from './path/to/auth-api.actions'

    function reducer(state, action: authApiActions.All) {
      switch(action.type) {
        case authApiActions.login.type: {...}
      }
    }
    ```
    It is still somewhat "1 export". I saw Martin's post. I cannot say that I agree with the final solution.
2. That's true that the style is different, but the goal is to keep it as close as possible, given that the style guide is driven by reasonable suggestions, right?
3. OK
4. OK (markdown won't let me skip these numbers)
5. It moves further from how `createSelector`s are organized (there's not containment object)
6. The **Action Union will hopefully be gone** soon-ish 🥳

Reply:
I don't know which I like the most.
I should be able to "refactor" the current app that I'm working on to use `createAction` during this week, and I'll have an opinion then 😄 

I was thinking about a 3rd option, which is don't use `unionize`.
- It isn't needed anymore in `Effects`
- We could use unionize directly in the reducers with only the use actions?
  If we take this a step further and if we would introduce a `createReducer` based on a specific reducer union, could we then check if all actions are handled within the reducer? For example if you would create a TODO reducer and add `CREATE TODO` and `REMOVE TODO`, if you wouldn't handle `REMOVE TODO` it would give errors at compile time.

Reply:
I'm kinda liking the thoughts I had here above.
As example, the collection reducer from the example app:

```ts
// define actions handled by this reducer
const actions = union({
  loadCollection,
  loadBooksSuccess,
  addBookSuccess,
  addBookFailure,
  removeBookSuccess,
  removeBookFailure,
});

export function reducer(state = initialState, act: Action): State {
  // type actions as mentioned in #951
  const action = act as typeof actions;

  switch (action.type) {
    case loadCollection.type: {
      return {
        ...state,
        loading: true,
      };
    }

    case loadBooksSuccess.type: {
      return {
        loaded: true,
        loading: false,
        ids: action.books.map(book => book.id),
      };
    }

    case addBookSuccess.type:
    case removeBookFailure.type: {
      if (state.ids.indexOf(action.book.id) > -1) {
        return state;
      }

      return {
        ...state,
        ids: [...state.ids, action.book.id],
      };
    }

    case removeBookSuccess.type:
    case addBookFailure.type: {
      return {
        ...state,
        ids: state.ids.filter(id => id !== action.book.id),
      };
    }

    default: {
      return state;
    }
  }
}
```

Reply:
What about going back to the regular const for the action creators and just using `ReturnType` whether there's one or many action creators? Keep the module imports and use the `Union` for reducers.

```ts
export type Union = ReturnType<
  | typeof loadCollection
  | typeof loadBooksSuccess
>;
```

Reply:
I'm fine with that as well. Let's just keep the `union` fn in the codebase as well (readjusted to take a tuple - I'll take care of that).
Again, hopefully the unions won't be needed for too much longer and are their for backwards-compatibility only :)